src/ssl_client.cpp

```diff
@@ -334,6 +334,9 @@ int start_ssl_client(sslclient_context *ssl_client, const char *host, uint32_t p
   log_v("Setting up IO callbacks...");
   mbedtls_ssl_set_bio(&ssl_client->ssl_ctx, ssl_client->client,
                       client_net_send, NULL, client_net_recv_timeout );
+  
+  log_v("Setting timeout...");
+  mbedtls_ssl_conf_read_timeout(&ssl_client->ssl_conf,  timeout);
```

Comment:
Have you been able to verify that the timeout is coming in as the timeout parameter in the callback to client_net_recv_timeout from mbedtls? e.g. maybe add this to first line of client_net_recv_timeout: 

`log_v("Timeout set to %d", timeout);`

Reply:
Yes I verified this, but removed the logging from the PR ;)